csbmysql/resource_binding_user_test.go

```diff
@@ -132,56 +140,90 @@ func checkUserIsCreated(username, password string, insecureUserConnection bool)
-		Expect(result.RowsAffected()).To(BeNumerically("==", 2))
+func validateBindingUser(dbUser *sql.DB) {
+	result, err := dbUser.Exec("insert into tasks(title, status, priority) values ('task', 2, 3), ('another task', 3, 4)")
+	Expect(err).NotTo(HaveOccurred())
+	Expect(result.RowsAffected()).To(BeNumerically("==", 2))
 
-		_, err = dbUser.Exec("drop table previous_table")
-		Expect(err).NotTo(HaveOccurred())
+	By("Reading and modifying existing data as the binding user")
+	rows, err := dbUser.Query(`select * from previous_table`)
+	Expect(err).NotTo(HaveOccurred())
 
-		By("Re-creating the pre-existing table")
+	Expect(rows.Next()).To(BeTrue())
+	var (
+		id    int
+		value string
+	)
+	err = rows.Scan(&id, &value)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(id).To(BeNumerically("==", 1))
+	Expect(value).To(Equal("value"))
+	Expect(rows.Next()).To(BeFalse())
 
-		_, err = dbUser.Exec("create table previous_table (pk int primary key not null auto_increment, value varchar(255) not null)")
-		Expect(err).NotTo(HaveOccurred())
+	result, err = dbUser.Exec(`insert into previous_table(pk, value) values (2, 'cheese')`)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(result.RowsAffected()).To(BeNumerically("==", 1))
 
-		result, err = dbUser.Exec(`insert into previous_table(pk, value) values (1, 'value')`)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(result.RowsAffected()).To(BeNumerically("==", 1))
+	result, err = dbUser.Exec(`update previous_table set value='new_value' where pk=1`)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(result.RowsAffected()).To(BeNumerically("==", 1))
 
-		return nil
-	}
+	result, err = dbUser.Exec(`delete from previous_table where pk in (1, 2)`)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(result.RowsAffected()).To(BeNumerically("==", 2))
+
+	_, err = dbUser.Exec("drop table previous_table")
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Re-creating the pre-existing table")
+
+	_, err = dbUser.Exec("create table previous_table (pk int primary key not null auto_increment, value varchar(255) not null)")
+	Expect(err).NotTo(HaveOccurred())
+
+	result, err = dbUser.Exec(`insert into previous_table(pk, value) values (1, 'value')`)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(result.RowsAffected()).To(BeNumerically("==", 1))
 }
 
-func checkUserIsDestroyed(username string) resource.TestCheckFunc {
+func validateReadOnlyBindingUser(dbUser *sql.DB) {
+	By("By validating that a readonly user can query data")
+	rows, err := dbUser.Query(`select * from previous_table`)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(rows.Next()).To(BeTrue())
+	var (
+		id    int
+		value string
+	)
+	err = rows.Scan(&id, &value)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(id).To(BeNumerically("==", 1))
+	Expect(value).To(Equal("value"))
+	Expect(rows.Next()).To(BeFalse())
+
+	By("By validating that a readonly user can't insert data")
+	_, err = dbUser.Exec(`insert into previous_table(pk, value) values (2, 'cheese')`)
+	Expect(err).To(HaveOccurred())
```

Comment:
thoughts: maybe to check the error message
    Expect(err).To(MatchError(ContainSubstring("INSERT command denied to user")))